packages/base/src/layerBrowser/layerBrowserDialog.tsx

```diff
@@ -47,17 +50,28 @@ export const LayerBrowserComponent = ({
     dialogHeader[0].setAttribute('style', 'padding: 0');
     dialog[0].classList.add('jgis-dialog-override');
 
-    sharedModel.layersChanged.connect(handleLayerChange);
+    model.sharedModel.layersChanged.connect(handleLayerChange);
 
     return () => {
-      sharedModel.layersChanged.disconnect(handleLayerChange);
+      model.sharedModel.layersChanged.disconnect(handleLayerChange);
     };
   }, []);
 
-  const handleLayerChange = () => {
-    // Get rid of the 'Layer' part of the name to match the names in the gallery
+  const handleLayerChange = (_, change: IJGISLayerDocChange) => {
+    const newLayers: string[] = [];
+    if (change.layerChange) {
+      change.layerChange.forEach(layer => {
+        newLayers.push(layer.id);
+      });
+
+      model.sharedModel.layersTree = [...model.getLayersTree(), ...newLayers];
```

Comment:
We were discussing this with @brichet earlier. 

We should update the SharedModel implementation so that it provides the proper APIs to add layers (and automatically add them to the tree).
Right now provides an `addLayer(id: string, value: IJGISLayer)` which only add it to the layer list but it should instead have an `addLayer(id: string, value: IJGISLayer, parentId: string, position: number)`.
Nicolas is looking into it.

Meanwhile Greg, I'd suggest sticking to the broken `addLayer` and not touching the tree for now so that you guys don't collide much.

Reply:
I wonder if this `addLayer` implementation should be in the `GISModel` instead of the `SharedModel`, since it is more a "high level" functionality.